lua/controller.lua

```diff
@@ -297,6 +297,79 @@ local function controller_insert_to_drawers(pos, stack)
 	return stack
 end
 
+--[[
+	Returns an array of drawers in the drawer network with their positions and slot
+	data. Slot data includes stored item, item count, and max count.
+
+	Results can be paginated by specifying an `offset` and a `max_count`.
+]]
+local function controller_get_network_info(pos, offset, max_count)
+	local found_drawers = {}
+	local index = index_drawers(pos)
+
+	-- Add each drawer in the network separately
+	-- Sort them by their positions to keep order for pagination
+	local drawer_positions = {}
+	local drawer_count = 0
+	local keys = {}
+	for _, drawer in pairs(index) do
+		local position = drawer.drawer_pos
+		local key = vector.to_string(position)
```

Comment:
`core.hash_node_position` should be faster to compute, is more memory efficient and also faster to sort.